### PR TITLE
Budget App (ZakaMoola)

### PR DESCRIPTION
**Pull request by HSMathebula**
### Presentation Video: [Vedio Link](https://www.loom.com/share/5ad703db885b4dd2905b22f69ed0f610)
In this PR I Added the following features:

- A splash page with the app's name and links to the signup and log-in pages were created.

- Sign up and log in to pages

  - The user can register in the app with their full name, email, and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)

  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page

  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page, the user can see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.
 
- "Add a new category" page

  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page

  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- Testing

  - Unit and integration tests for Group and Expenses page.
  
**Thank you for your review in advance!**

### CODE REVIEW

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Happen, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- professional-looking Readme file.
- linters are passing

## Required Changes ♻️
- [ ] there is no link to the video presentation available, please include it and request for review again.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HSMathebula** [comment]
@mikethreels Thank you

File: README.md [hunk 1]
```diff
@@ -1,24 +1,73 @@
...
+
+👤 **Happen Mathebula**
+
+- GitHub: [@HSMathebula](https://github.com/HSMathebula)
+- Twitter: [@happen_sipho](https://twitter.com/HappenSipho)
+- LinkedIn: [LinkedIn](www.linkedin.com/in/happen-sipho-mathebula)
+
+
+## Attributions
+
```
Comment (Whoistolu) [inline README.md:53]: - [ ] [OPTIONAL] Great implementation @HSMathebula 👍🏽  however, for proper project documentation, kindly add the link for the project deployment and the project video presentation to the README file. It is also good as you have attached the links to the PR description.

**Review by Whoistolu — APPROVED** [review]
### Hi @HSMathebula 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**